### PR TITLE
Issue/SSR static page 

### PR DESCRIPTION
**Pull request by owensmooth**
related issue: https://github.com/shiftcommerce/shift-front-end-react/issues/890

SSR was broken on static pages because we set the state to loading for both server side and client side. The fix was to check if we are server side and if so don't display the loading spinner.

I've also removed a lot of methods in static-page that aren't needed, which seems to have improved performance. 

I have also added a global state for loading, the action has been added into api-client so when an API call is fired we set loading to true, when we get the response back or the API call errors we set loading to false. This doesn't have to be used for every API call if you don't want to, it has been implemented so it only works when you pass dispatch through with the API call. 
**This will need a follow-up PR to adjust the current way we handle loading states in pages**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ module.exports = {
     react: 'commonjs react',
     'react-dom': 'commonjs react-dom',
     '@shiftcommerce/shift-react-components': 'commonjs @shiftcommerce/shift-react-components',
+    '@shiftcommerce/shift-node-api': 'commonjs @shiftcommerce/shift-node-api',
```
Comment (owensmooth) [inline webpack.config.js]: Webpack was complaining about shift-node-api, this fixed the error

File: src/lib/api-client.js [hunk 2]
```diff
@@ -4,40 +4,58 @@ import axios from 'axios'
 
 import Config from '../lib/config'
 
+import { toggleLoading } from '../actions/global-actions'
+
+// Check to see if dispatch is passed through,
+// if it is dispatch this action, otherwise dont do anything.
```
Comment (cbarratt) [inline src/lib/api-client.js]:
```suggestion
// if so, dispatch `toggleLoading`, otherwise don't do anything.
```

File: src/pages/static-page.js [hunk 3]
```diff
@@ -8,60 +8,31 @@ import ApiClient from '../lib/api-client'
...
-  }
-
   /**
-   * Request the page from the API
-   * @param  {Number} id
-   * @return {Object} API response or error
-   */
-  static async fetchPage (id) {
+ * Request the page from the API
+ * @param  {Number} id
```
Comment (cbarratt) [inline src/pages/static-page.js]: Need to add the dispatch as a param here

File: src/actions/global-actions.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import * as types from './action-types'
+
+export function toggleLoading(state) {
+  return {
+    type: types.TOGGLE_LOADING,
+    loading: state
+  }
+}
```
Comment (mufudzimasaire) [inline src/actions/global-actions.js]: Minor - new line

File: src/reducers/global-reducer.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  switch(action.type) {
+    case types.TOGGLE_LOADING:
+      return Object.assign({}, state, {
+        loading: action.loading
+      })
+
+    default:
+      return state;
+  }
+}
```
Comment (mufudzimasaire) [inline src/reducers/global-reducer.js]: newline

File: test/reducers/global-reducer.test.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  expect(globalReducer({ loading: false }, trueAction)).toEqual({
+    loading: true
+  })
+
+  expect(globalReducer({ loading: true }, falseAction)).toEqual({
+    loading: false
+  })
+})
```
Comment (mufudzimasaire) [inline test/reducers/global-reducer.test.js]: newline

**Pull request merged**